constrain-lib/src/main/resources/schema/dependency-constraints-schema.json

```diff
@@ -0,0 +1,123 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "type": "object",
+  "title": "Dependency Constraints Schema.",
+  "required": [
+    "dependencyConstraints"
+  ],
+  "properties": {
+    "dependencyConstraints": {
+      "$ref": "#/$defs/dependencyConstraints"
+    }
+  },
+  "additionalProperties": false,
+  "$defs": {
+    "dependencyConstraints": {
+      "type": "array",
+      "title": "Constraints",
+      "description": "A collection of Dependency Constraints.",
+      "items": {
+        "$ref": "#/$defs/dependencyConstraint"
+      }
+    },
+    "dependencyConstraint": {
+      "type": "object",
+      "title": "Constraint",
+      "description": "A single Dependency Constraint.",
+      "required": [
+        "groupId",
+        "name",
+        "suggestedVersion",
+        "because"
+      ],
+      "properties": {
+        "groupId": {
+          "type": "string",
+          "title": "Group",
+          "description": "The group of the dependency.",
+          "examples": [
+            "com.example"
+          ]
+        },
+        "name": {
+          "type": "string",
+          "title": "Name",
+          "description": "The name of the dependency.",
```

Comment:
I would clarify that this is the `artifactId` and not a nice name, e.g., `spring-core` vs `Spring Framwork Core`